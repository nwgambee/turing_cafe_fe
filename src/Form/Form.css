.res-form {

}

input {
  padding: 5px;
  margin: 5px;
}

.make-res {
  color: white;
  background-color: teal;
  font-size: 1em;
  padding: 4px;
  border-radius: 3px;
  cursor: pointer;
}
