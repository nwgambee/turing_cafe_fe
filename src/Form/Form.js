import React, { Component } from 'react';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }
  handleChange = (event) => {
    console.log(event);
  }

  render() {
    return (
      <form className='res-form'>
        <input
          placeholder='Name'
          value={}
          onChange={this.handleChange}
        />
        <input
          placeholder='Date'
          value={}
          onChange={this.handleChange}
        />
        <input
          placeholder='Time'
          value={}
          onChange={this.handleChange}
        />
        <input
          placeholder='# of Guests'
          value={}
          onChange={this.handleChange}
        />
    )
  }
}


export default Form;
