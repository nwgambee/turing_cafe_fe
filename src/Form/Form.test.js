import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import {shallow} from 'enzyme';

it('should match the snapshot', () => {
  let wrapper = shallow(<Form addReservation={jest.fn()} />)
  expect(wrapper).toMatchSnapshot();
})
