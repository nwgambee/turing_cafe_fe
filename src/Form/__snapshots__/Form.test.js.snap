// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<form
  className="res-form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="# of Guests"
    value=""
  />
  <button
    className="make-res"
    onClick={[Function]}
  >
    Make Reservation
  </button>
</form>
`;
